azurerm/internal/services/postgres/postgresql_flexible_server_data_source.go

```diff
@@ -0,0 +1,146 @@
+package postgres
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/postgresql/mgmt/2020-02-14-preview/postgresqlflexibleservers"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/location"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourcePostgresqlFlexibleServer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPostgresqlFlexibleServerRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"storage_mb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"delegated_subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"backup_retention_days": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"cmk_enabled": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"high_availiblity_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"public_network_access_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"standby_availability_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceArmPostgresqlFlexibleServerRead(d *schema.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Postgres.FlexibleServersClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	id := parse.NewFlexibleServerID(subscriptionId, resourceGroup, name).ID()
+
+	resp, err := client.Get(ctx, resourceGroup, name)
```

Comment:
Again, can we not make id a string, and use the id struct throughout the function
```suggestion
	id := parse.NewFlexibleServerID(subscriptionId, resourceGroup, name).ID()

	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
```